src/main/java/com/minecolonies/coremod/network/messages/RecallCitizenMessage.java

```diff
@@ -111,20 +112,25 @@ public void messageOnServerThread(final RecallCitizenMessage message, final Enti
                     else if (optionalEntityCitizen.get().getTicksExisted() == 0)
                     {
                         final AbstractEntityCitizen oldCitizen = optionalEntityCitizen.get();
-                        final List<AbstractEntityCitizen> list = player.getServerWorld().getLoadedEntityList().stream()
-                                                                   .filter(e -> e instanceof AbstractEntityCitizen)
-                          .filter(e -> e.equals(oldCitizen))
-                                                                   .map(e -> (AbstractEntityCitizen) e)
-                          .collect(Collectors.toList());
+                        final ICitizenData oldCitizenData = oldCitizen.getCitizenData();
 
-                        if (list.isEmpty())
+                        if (oldCitizenData != null)
                         {
-                            citizenData.setCitizenEntity(null);
-                            citizenData.updateCitizenEntityIfNecessary();
-                        }
-                        else
-                        {
-                            citizenData.setCitizenEntity(list.get(0));
+
+                            final List<AbstractEntityCitizen> list = new ArrayList<>(player.getServerWorld()
+                                                                                       .getEntities(AbstractEntityCitizen.class,
+                                                                                         entityCitizen -> entityCitizen.getCitizenColonyHandler().getColonyId() == colony.getID()
+                                                                                                            && entityCitizen.getCitizenData().getId() == oldCitizenData.getId()));
```

Comment:
formatting

Reply:
wdym?

Reply:
That's the best way to format it imo

Reply:
like the amount of spaces is disgusting, I usually use one indent layer for multi-line expressions, someone does two indent layers to distinguish between blocks and multi-line

Reply:
I usually go with 1 extra indent for the lambda, and then a new one for each logical subexpression.

Reply:
The important thing is that it is readable. The lambda is the central point of the expression, so it has to be understandable where it starts and what the conditions are

Reply:
ah I see, you just use two spaces instead of four
what I was writing about is the number of spaces when the line is shifted to the new indent layer according to the previous line

Reply:
Oh, that's what Intellij does for me on Linux =D